src/stores/useSocialMediaStore/useSocialMediaStore.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { act, renderHook } from '@testing-library/react';
+
+import * as useSocialMediaStoreHook from '~stores/useSocialMediaStore/useSocialMediaStore';
+
+vi.mock('~services/api/social-media/social-media', () => ({
+  SocialMediaService: {
+    fetch: vi.fn().mockResolvedValue([
+      { icon: 'discord-icon', id: 'DISCORD_EXAMPLE_ID', name: 'Discord' },
+      { icon: 'twitter-icon', id: 'TWITTER_EXAMPLE_ID', name: 'Twitter' },
+    ]),
+  },
+}));
+
+vi.mock('~services/api/accounts/accounts', () => ({
+  AccountsService: {
+    fetchAll: vi.fn().mockResolvedValue([
+      {
+        avatar: 'https://example.com/image1.jpg',
+        expiresAt: 'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        generatedAt:
+          'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        id: 1,
+        socialMediaId: 'DISCORD_EXAMPLE_ID',
+        token: 'DISCORD_EXAMPLE_TOKEN_1',
+        userName: 'Discord User 1',
+        valid: true,
+      },
+      {
+        avatar: 'https://example.com/image2.jpg',
+        expiresAt: 'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        generatedAt:
+          'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        id: 14,
+        socialMediaId: 'TWITTER_EXAMPLE_ID',
+        token: 'TWITTER_EXAMPLE_TOKEN_14',
+        userName: 'Twitter User 14',
+        valid: true,
+      },
+    ]),
+  },
+}));
+
+describe('useSocialMediaStore Hook', () => {
+  it('Checks if the fetchedSocialMediasMap is correctly set', async () => {
+    const { result } = renderHook(() =>
+      useSocialMediaStoreHook.useSocialMediaStore()
+    );
+
+    await act(async () => {
+      await result.current.getAllAccounts();
+    });
+
+    const expectedMap = new Map();
+    expectedMap.set('DISCORD_EXAMPLE_ID', {
+      icon: 'discord-icon',
+      id: 'DISCORD_EXAMPLE_ID',
+      name: 'Discord',
+    });
+    expectedMap.set('TWITTER_EXAMPLE_ID', {
+      icon: 'twitter-icon',
+      id: 'TWITTER_EXAMPLE_ID',
+      name: 'Twitter',
+    });
+
+    expect(result.current.socialMedias).toEqual(expectedMap);
+  });
+
+  it('Checks if the function correctly handles input and produces expected output', async () => {
+    const updatedAccounts = [
+      { id: 1, socialMediaId: 'DISCORD_EXAMPLE_ID', valid: false },
+      { id: 14, socialMediaId: 'TWITTER_EXAMPLE_ID', valid: false },
+    ];
+
+    const { result } = renderHook(() =>
+      useSocialMediaStoreHook.useSocialMediaStore()
+    );
+
+    await act(async () => {
+      await result.current.getAllAccounts();
+    });
+
+    expect(result.current.accounts.data).toEqual(
+      expect.arrayContaining(
+        updatedAccounts.map((account) => expect.objectContaining(account))
+      )
```

Comment:
Não podemos simplesmente comparar assim?

```suggestion
      updatedAccounts
```

Reply:
Nao, pois o primeiro teste passa se os objetos corretos estão presentes, independentemente da ordem ou de propriedades extras. Enquanto o segundo teste só passa se o array for exatamente igual ao esperado, incluindo a ordem e sem propriedades extras. Quando mudei o segundo teste falhou, acreido que e' pq a ordem esteja diferente.